lib/sidekiq/cron/job.rb

```diff
@@ -26,11 +26,17 @@ def initialize input_args = {}
         @fetch_missing_args = true if @fetch_missing_args.nil?
 
         @name = args["name"]
-        @namespace = args["namespace"] || Sidekiq::Cron.configuration.default_namespace
         @cron = args["cron"]
         @description = args["description"] if args["description"]
         @source = args["source"] == "schedule" ? "schedule" : "dynamic"
 
+        default_namespace = Sidekiq::Cron.configuration.default_namespace
+        @namespace = args["namespace"] || default_namespace
+        if Sidekiq::Cron::Namespace.available_namespaces_provided? && !Sidekiq::Cron::Namespace.all.include?(@namespace) && @namespace != default_namespace
+          Sidekiq.logger.warn { "Cron Jobs - unexpected namespace #{@namespace} encountered. Assigning to default namespace." }
+          @namespace = Sidekiq::Cron.configuration.default_namespace
```

Comment:
Maybe this line can be something like: 

```suggestion
          @namespace = default_namespace
```

Reply:
Yep, thanks